### ISSUE TITLE
Button to show enabled addons

### ISSUE THREAD

**Issue by Samq64** [ScratchAddons/ScratchAddons#5068]
### READ BEFORE CREATING!

- [X] I understand

### Why this would be helpful/solve a problem

It would be useful to have the list of enabled addons in bug reports, but the json file is too large for that.

### What this change does

Adding a button somewhere to open https://scratchaddons.github.io/list-of-enabled-addons/ with the user's enabled addons and then just pasting the URL with the generated number into bug reports.

### Possible alternatives

Using the json file or upgrading the webpage to also list other information like Browser version, operating system, and Scratch Addons version and just get the user to paste that URL into the bug report without anything else.

**Comment by WorldLanguages** [comment]
> upgrading the webpage to also list other information like Browser version, operating system, and Scratch Addons version

Something like this could be added to "more settings" I'd say.
It would show (and potentially copy to clipboard) a string with:
1. SA version_name
2. Git head hash, if available (I think we could fetch `/.git/FILENAME` from JS to get this)
3. User agent (browser and OS)
4. Direct link to list-of-enabled-addons with the corresponding hash

The bug report form could then have a single textbox where we ask users to paste the generated string.

**Comment by DNin01** [comment]
I don't think we necessarily have to use a website to do this, we could just have a button in More Settings (in a Debugging category or something like that) to show a list of enabled addons and maybe some other information in a popup that could then be copied in one piece and pasted into a section of the Bug Report issue form.

We could also just ask users in the Bug Report form to provide their settings file, which is already accessible from More Settings, but that might be harder to utilize with so much information included.

**Comment by WorldLanguages** [comment]
> I don't think we necessarily have to use a website to do this, we could just have a button in More Settings (in a Debugging category or something like that) to show a list of enabled addons and maybe some other information in a popup that could then be copied in one piece and pasted into a section of the Bug Report issue form.

The text would be too long. If we somehow compressed it, then inexperienced users wouldn't know what to do with it.

In my opinion, using URLs is useful because:
1. Users already know how to copy and paste URLs
2. Users are aware of what information they are sharing
3. We already use these types of URLs in the current, closed feedback system

**Comment by DNin01** [comment]
> > I don't think we necessarily have to use a website to do this, we could just have a button in More Settings (in a Debugging category or something like that) to show a list of enabled addons and maybe some other information in a popup that could then be copied in one piece and pasted into a section of the Bug Report issue form.
> 
> The text would be too long. If we somehow compressed it, then inexperienced users wouldn't know what to do with it.
> 
> In my opinion, using URLs is useful because:
> 
> 1. Users already know how to copy and paste URLs
> 2. Users are aware of what information they are sharing
> 3. We already use these types of URLs in the current, closed feedback system

I would assume the text could be something easy to read and understand, like:
```
SA Version: 1.29.0
OS: Windows 10 / Browser: Microsoft Edge 107
Enabled Addons: clones, pause, (...)
```

**Comment by WorldLanguages** [comment]
@DNin01 The list of enabled addons would be long for many users, and there would be no reasonable way to add setting values there.

**Comment by DNin01** [comment]
My idea would be to encode the list of enabled addons as part of a URL and then visiting that URL would lead to a page that decodes the part of the URL that encodes enabled addons?

Like, let's say if the addon `scratch-messaging` had a numerical ID of 1 and the user had only that addon enabled. Upon clicking a button to export a list of enabled addons, they would be sent to `https://somewebsite/1` which would then decode the "1" into `scratch-messaging` and display it on the page. Easy for everyone to make use of.

This would mean that we wouldn't have to store any data in servers and we wouldn't have to worry about privacy concerns regarding data retention because the data is stored in the URL (that's like if a file name stored data and the file itself was just a script to make use of its file name).

Related repo: https://github.com/ScratchAddons/list-of-enabled-addons

**Comment by WorldLanguages** [comment]
Yeah, the hardest part is trying to encode the addonSettings into a few characters so that it can fit in a reasonably-sized URL. I guess we could save a few bytes by omitting settings that are set to their default value.

**Comment by DNin01** [comment]
> Yeah, the hardest part is trying to encode the addonSettings into a few characters so that it can fit in a reasonably-sized URL. I guess we could save a few bytes by omitting settings that are set to their default value.

For boolean settings, we could include the encoded name of the setting (which could probably be a number) if the setting was not set to the default, so we wouldn't have to encode "true" or "false". We could also exclude table settings since those can be whatever the user wants, that would drastically cut down on the size.

**Comment by DNin01** [comment]
While numbers don't take up much space, using numbers for basically everything would mean we'd need a way to keep the encoded data forwards compatible, because if we don't store numerical IDs for settings and they get rearranged, it could misinterpret the encoded settings.

**Comment by DNin01** [comment]
I just made a script which you can inject into the "View settings file" page. It will output the addons you have enabled so you can easily provide them in bug reports.

https://gist.github.com/DNin01/37f9f81404dcdfffd22aefabd9dd4e84
